prow/cmd/branchprotector/protect.go

```diff
@@ -170,7 +163,7 @@ func (p *Protector) Protect() {
 
 	// Scan the branch-protection configuration
 	for orgName, org := range bp.Orgs {
-		if err := p.UpdateOrg(orgName, org, bp.Protect); err != nil {
+		if err := p.UpdateOrg(orgName, org, bp.Protect != nil); err != nil {
```

Comment:
I don't really understand this cascading `*bool` pass -- why do we try to resolve this at the call-sites for all of the `p.Update*` methods? The `(c *Config) GetBranchProtection` call should be able to resolve in one place for every `(org,repo,branch)` tuple whether or not that specific triad of identifiers requires protection based on explicit config and the parent-child relationships there. I see that with this approach we only conditionally call the GitHub API to get org/repo/branch lists but with the cache that should be basically free as those items really do not change often. Maybe it would be better just to get all the info and filter in one place, not in both?

Reply:
This is passing a `bool`. The `*bool` was silly as we only check whether its nil throughout its lifetime (and then recompute the value in `GetBranchProtection`), and yes we use it exclusively to determine whether we need to read all branches/repos or just defined ones.

The cache doesn't exist yet, does it? It also feels wrong to me to read every branch in the org just because I want to protect a couple branches in a couple repos.

Would you be happier if I do not change the signature of `Update*` methods (so keep the `*bool`? The PR is fine without it.

Reply:
Yeah I guess if we really want to optimize for GitHub API reads here this approach is fine. Was just a bit hard to follow at first.